packages/pluginmanager/src/model.ts

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import type { JupyterLab } from '@jupyterlab/application';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+import { VDomModel } from '@jupyterlab/ui-components';
+import { ISignal, Signal } from '@lumino/signaling';
+import { PromiseDelegate } from '@lumino/coreutils';
+import {
+  ITranslator,
+  nullTranslator,
+  TranslationBundle
+} from '@jupyterlab/translation';
+
+import { PluginInUseMessage, PluginRequiredMessage } from './dialogs';
+
+/**
+ * The server API path for querying/modifying available plugins.
+ */
+const PLUGIN_API_PATH = 'lab/api/plugins';
```

Comment:
Wondering if some subset of the functionalities could still work without making requests to a server implementing this endpoint?

Reply:
Use cases are JupyterLite or other alternative lab based apps (similar to the ones in `examples`), to be able to easily see the list of plugins available in a given app.

Reply:
Yes, absolutely. See https://github.com/jupyterlab/jupyterlab/pull/14536/files/bfb21875863808c5b057adc2512c0297da96c760#r1311327774